### PR TITLE
Some network changes (  missing push event at IP change,  Active scan, connect best signal AP ) 

### PR DESCRIPTION
**Pull request by KipK**
Some fixes & improvement:

missing push event at ip change
set wifi scan to active ( solve strange bug ) 
force wifi to conenct to AP with best signal, fixes https://github.com/OpenEVSE/ESP32_WiFi_V4.x/issues/385

### CODE REVIEW

**Comment by jeremypoulter** [comment]
If this is needed you also need to renew the DHCP, some routers expose the hostname via 'normal' DNS, so probably need to restart the whole network stack

**Comment by KipK** [comment]
I got this issue during wizard if user set the hostname after setting the wifi network, then at the end of wizard it redirects to the new hostname but mDNS hasn't been updated on the network. 
I use hostname for now, I never receive the new IP while in AP+STA mode, as it fails to connect until AP mode is OFF.
I'lm afraid reloading the whole network stack would disconnect user from AP, wouldn't ?

**Comment by KipK** [comment]
This also add missing event_send ip & connection status in wifiOnStationModeGotIP

However most of the time I don't receive it, I'm kicked out of the AP when saving the new WIFI/pass

**Comment by jeremypoulter** [comment]
Thanks @KipK, it is not the DNS server on the ESP that needs restarting, that is only used in AP mode. It is that the ESP needs to renew its IP settings from the router, the host name is contained in the DHCP message and this is then used by some routers to in the router's DNS, eg I can use `openevse-xxxx.lan` no resolve the OpenEVSE IP, much faster, works on more devices and more reliable than mDNS.

IMHO I think it may just be a case of restarting the networking (both wired and wireless), unless there is a handy method to just enew the IP address.

**Comment by jeremypoulter** [comment]
> force wifi to conenct to AP with best signal

How did you do this?

**Comment by jeremypoulter** [comment]
Nevermind, found it.... If I had known it was that simple would have made that change ages ago....

**Comment by KipK** [comment]
> Nevermind, found it.... If I had known it was that simple would have made that change ages ago....

found that while debugging the wifi issue , was worth trying :)

**Comment by KipK** [comment]
> 

what would be the best way to restart this without restartint the AP mode ?

**Comment by jeremypoulter** [comment]
If you are in AP mode, just restarting the mDNS is sufficient. When connected to a proper network (either via WiFi or Wired) that is when you would need to restart the network.

That being said the cases when you would be in AP mode and change the hostname will be slim I would have thought...

**Comment by KipK** [comment]
> That being said the cases when you would be in AP mode and change the hostname will be slim I would have thought...

on the wizard in AP mode, the Hostname can be set.
I'll add a condition for AP mode only thx.

**Comment by KipK** [comment]
this should be good like this.

File: src/net_manager.cpp [hunk 1]
```diff
@@ -73,6 +74,14 @@ void NetManagerTask::begin()
   }
 }
 
+void NetManagerTask::end()
+{
+  if(NULL != _instance)
+  {
+    MicroTask.stopTask(_instance);
```
Comment (jeremypoulter) [inline src/net_manager.cpp:81]: You will need to explicitly shutdown the networking (after stopping the task), there is no equivalent of setup in MicroTasks

Comment (KipK) [reply src/net_manager.cpp:81]: what would be the method to shutdown ETH ?

**Comment by KipK** [comment]
I've removed the hostanme change triggering dns restart and co. 
Dunno how to do it properly for now, i get this out of this PR.

**Comment by jeremypoulter** [comment]
Yeah, maybe a good idea. I can have a look, I think I also need to look at changing WiFi credentials as well

**Comment by KipK** [comment]
I've found Glynn problem with captive portal in AP+STA, it nows works ok.

**Comment by KipK** [comment]
https://github.com/OpenEVSE/ESP32_WiFi_V4.x/commit/bab6d5a43f36aa0c58d02244945d4e9dd26c8ed6 should now allows you want you want

**Review by jeremypoulter — APPROVED** [review]

**Pull request merged**